web/app/js/components/util/ApiHelpers.jsx

```diff
@@ -3,6 +3,39 @@ import { Link } from 'react-router-dom';
 import React from 'react';
 import 'whatwg-fetch';
 
+const checkFetchOk = resp => {
+  if (!resp.ok) {
+    throw Error(resp.statusText);
+  } else {
+    return resp;
+  }
+};
+
+// makeCancelable from @istarkov
+// https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html
+const makeCancelable = (promise, onSuccess) => {
+  let hasCanceled_ = false;
```

Comment:
Just curious about what this is used for. Is function created for testability?

Reply:
this is the function that wraps a promise and makes it a cancelable promise. there's no way to actually cancel a fetch request (currently), but we'd like a way to stop caring about the result of an unresolved promise. so we wrap a regular promise in an object, such that we now have:
```
{
  promise: ... // our original promise with some handlers,
  cancel: ... // a method allowing us to say when we don't care about the results of the resolve promise any more
}
```

On line 55, you can see we're using this to wrap our fetch calls:
```
return makeCancelable(fetch(path), r => r.json());
```